### PR TITLE
refactor: Refactored few files to make the code better.

### LINKED ISSUE #27217: Remove unnecessary-double-cast iterables.
The following optimizations can be made to make the code better:
It's unnecessary to double-cast or double-process iterables by wrapping the listed functions within an additional `list`, `set`, `sorted`, or `tuple` call. Doing so is redundant and can be confusing for readers.

There is no reason to use a `list comprehension` if the result is `immediately unpacked`. Instead, use a `generator` expression, which is more efficient as it avoids allocating an intermediary list.

I used [`ruff`](https://docs.astral.sh/ruff/) linter to identify these.

### PR DESCRIPTION
**Pull request by Sai-Suraj-27**
# PR Description
The following optimizations can be made to make the code better:
It's unnecessary to double-cast or double-process iterables by wrapping the listed functions within an additional `list`, `set`, `sorted`, or `tuple` call. Doing so is redundant and can be confusing for readers.

There is no reason to use a `list comprehension` if the result is `immediately unpacked`. Instead, use a `generator` expression, which is more efficient as it avoids allocating an intermediary list.

I used [`ruff`](https://docs.astral.sh/ruff/) linter to identify and fix these.

## Related Issue
Closes #27217

## Checklist

- [ ] Did you add a function?
- [ ] Did you add the tests?
- [ ] Did you run your tests and are your tests passing?
- [x] Did pre-commit not fail on any check?
- [ ] Did you follow the steps we provided?

### Socials
https://twitter.com/Sai_Suraj_27

### CODE REVIEW

File: ivy_tests/test_ivy/test_functional/test_core/test_device.py
Comment (KareemMAX) [inline ivy_tests/test_ivy/test_functional/test_core/test_device.py:292]: Are you sure this will pass the equality test?

File: ivy_tests/test_ivy/test_functional/test_core/test_device.py
Comment (KareemMAX) [inline ivy_tests/test_ivy/test_functional/test_core/test_device.py:311]: Are you sure this will pass the equality test?

File: ivy_tests/test_ivy/test_functional/test_core/test_dtype.py
Comment (KareemMAX) [inline ivy_tests/test_ivy/test_functional/test_core/test_dtype.py:658]: Are you sure this will pass the equality test?

File: ivy_tests/test_ivy/test_functional/test_core/test_dtype.py
Comment (KareemMAX) [inline ivy_tests/test_ivy/test_functional/test_core/test_dtype.py:670]: Are you sure this will pass the equality test?

**Review by KareemMAX — CHANGES_REQUESTED** [review]
Hey @Sai-Suraj-27,

Pointed out some small issues that I'm not certain of. Also @vedpatwardhan can you take a look before merging?

File: ivy_tests/test_ivy/test_functional/test_core/test_device.py
Comment (Sai-Suraj-27) [reply ivy_tests/test_ivy/test_functional/test_core/test_device.py:292]: Yes, `exp` and `res` are already iterable objects, removing the `tuple()` should not change anything because the `sorted` function acts the same on any iterable. So, I think it will pass the equality test.

**Comment by AnnaTz** [comment]
Hi @Sai-Suraj-27, I don't see any issue with these changes, but I think we should test everything to make sure no failures are caused in practice.
I think @vedpatwardhan should take a look into this as well before we approve the merge.

**Review by vedpatwardhan — APPROVED** [review]
lgtm! Feel free to merge, thanks @Sai-Suraj-27 😄

**Comment by Sai-Suraj-27** [comment]
> Hi @Sai-Suraj-27, I don't see any issue with these changes, but I think we should test everything to make sure no failures are caused in practice. Have you tried running the tests of the functions you modified? I think @vedpatwardhan should take a look into this as well before we approve the merge.

Yes, I ran tests for these 2 files `ivy_tests\test_ivy\test_functional\test_core\test_dtype.py`, `ivy_tests/test_ivy/test_functional/test_core/test_device.py`, and tests of this function that I modified `def zero_fraction`.

**None of the tests where I made changes are failing.** So, I guess there is no problem 👍.


image


image

**Pull request merged**
